man/man8/zfsprops.8

```diff
@@ -855,11 +855,11 @@ is selected, compression will explicitly check for blocks consisting of only
 zeroes (the NUL byte).  When a zero-filled block is detected, it is stored as
 a hole and not compressed using the indicated compression algorithm.
 .Pp
-Any block being compressed must be no larger than 7/8 of its original size
-after compression, otherwise the compression will not be considered worthwhile
-and the block saved uncompressed. Note that when the logical block is less than
-8 times the disk sector size this effectively reduces the necessary compression
-ratio; for example 8k blocks on disks with 4k disk sectors must compress to 1/2
+Block must be compressed by at least max ashift of pool vdevs, otherwise
+the compression will not be considered worthwhile and the block saved
+uncompressed. Note that when the logical block is less than 8 times the disk
+sector size this effectively reduces the necessary compression ratio;
+for example 8k blocks on disks with 4k disk sectors must compress to 1/2
 or less of their original size.
```

Comment:
I took a stab at re-explaining these concepts, although I feel we could also just drop this paragraph, as it is fairly deep in the implementation details.

```suggestion
All blocks are allocated as a whole number of sectors (chunks of 2^ashift bytes, e.g. 512B or 4KB).  Compression may result in a non-sector-aligned size, which will be rounded up to a whole number of sectors.  If compression saves less than one whole sector, the block will be stored uncompressed.  Therefore, blocks whose logical size is a small number of sectors will experience less compression ratio  (e.g. for recordsize=16K with 4K sectors, which have 4 sectors per block, compression needs to save at least 25% to actually save space on disk).
```